crates/proto/src/xfer/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -70,12 +70,20 @@
 use crate::{
     op::{Query, ResponseCode},
     rr::{
-        dnssec::{rdata::NSEC3, Proof},
+        dnssec::{rdata::NSEC3, Nsec3HashAlgorithm, Proof},
         Name, Record, RecordType,
     },
     xfer::dnssec_dns_handle::is_dnssec,
 };
 
+fn nsec3hash(name: &Name, salt: &[u8], iterations: u16) -> Vec<u8> {
+    Nsec3HashAlgorithm::SHA1
+        .hash(salt, name, iterations)
+        .expect("SHA1 Hashing of a domain name failed")
```

Comment:
Please make this explanation about why it cannot fail, instead of just saying it failed.